### PR TITLE
mas on raspberry pi

### PR DESCRIPTION
**Pull request by chschrae**
## Purpose
This PR will automatically download MAS in the raspberry pi build scripts.
I have also added config options necessary for MAS use on linux.
There is a separate build script defining a macro that will decide which code is used for creating the dialog service connector.

Readme for the raspberry pi has also been updated.

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x] Feature
[ ] Code style update (formatting, local variables)
[ x] Refactoring (no functional changes, no api changes)
[x ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

build it on raspberry pi and see that it works.

### CODE REVIEW

File: clients/cpp-console/configs/micConfig.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+            "yCoord": 0,
+            "zCoord": 0
+        },
+        {
+            "xCoord": 0,
+            "yCoord": 0,
+            "zCoord": 0
+        }
+    ]
+}
```
Comment (dargilco) [inline clients/cpp-console/configs/micConfig.json:16]: Can you give a more realistic example here? (where mic1 does not equal mic2 coordinates). Something from a mic device.

File: clients/cpp-console/docs/RaspberryPi.md [hunk 2]
```diff
@@ -46,8 +46,35 @@ There are many ways to do development on a Raspberry pi. It may be useful to tak
...
-  sh ./buildArm32.sh
+  ./buildArm32Linux.sh
   ```
+  
+## Build directly on Linux arm32 with Microsoft Audio Stack
+
+Microsoft Audio Stack can be used to enhance the experience.
+
+Benefits include:
+* Software Audio Echo Cancellation
```
Comment (dargilco) [inline clients/cpp-console/docs/RaspberryPi.md:57]: "Acoustic Echo Cancellation"

File: clients/cpp-console/docs/RaspberryPi.md [hunk 3]
```diff
@@ -46,8 +46,35 @@ There are many ways to do development on a Raspberry pi. It may be useful to tak
...
+* Beamforming
+* Noise Suppression
+* Automatic Gain Control
+* Dereverberation
```
Comment (dargilco) [inline clients/cpp-console/docs/RaspberryPi.md:61]: I don't think dereverberation is part of MAS. Or maybe this got added recently? Is this from MAS documentations?

File: clients/cpp-console/docs/RaspberryPi.md [hunk 4]
```diff
@@ -46,8 +46,35 @@ There are many ways to do development on a Raspberry pi. It may be useful to tak
 
   ```sh
   cd scripts
-  sh ./buildArm32.sh
+  ./buildArm32Linux.sh
   ```
+  
+## Build directly on Linux arm32 with Microsoft Audio Stack
+
+Microsoft Audio Stack can be used to enhance the experience.
```
Comment (dargilco) [inline clients/cpp-console/docs/RaspberryPi.md:54]: "Microsoft Audio Stack (MAS)"

File: clients/cpp-console/docs/RaspberryPi.md [hunk 5]
```diff
@@ -46,8 +46,35 @@ There are many ways to do development on a Raspberry pi. It may be useful to tak
...
+
+Benefits include:
+* Software Audio Echo Cancellation
+* Beamforming
+* Noise Suppression
+* Automatic Gain Control
+* Dereverberation
+
+In order to use Microsoft Audio Stack you must specify in the config file the appropriate values. Below are the ones that we used on our pi:</br>
```
Comment (dargilco) [inline clients/cpp-console/docs/RaspberryPi.md:63]: pi? You mean Raspberry Pie?

File: clients/cpp-console/docs/RaspberryPi.md (hunk 3 above)
Comment (chschrae) [reply clients/cpp-console/docs/RaspberryPi.md:61]: I got this list off a slide in the deck Ian sent me so I guess it is?

File: clients/cpp-console/docs/RaspberryPi.md (hunk 5 above)
Comment (mswellsi) [reply clients/cpp-console/docs/RaspberryPi.md:63]: Raspberry Pi see https://www.raspberrypi.org/

File: clients/cpp-console/docs/RaspberryPi.md (hunk 3 above)
Comment (mswellsi) [reply clients/cpp-console/docs/RaspberryPi.md:61]: Yes, it is one of the improvements in MAS over NUI Audio.

File: clients/cpp-console/configs/micConfig.json (hunk 1 above)
Comment (mswellsi) [reply clients/cpp-console/configs/micConfig.json:16]: See here https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/how-to-devices-microphone-array-configuration for some examples

File: clients/cpp-console/docs/RaspberryPi.md (hunk 2 above)
Comment (chschrae) [reply clients/cpp-console/docs/RaspberryPi.md]: fixed

File: clients/cpp-console/configs/micConfig.json (hunk 1 above)
Comment (chschrae) [reply clients/cpp-console/configs/micConfig.json]: done and added link in readme to the docs page about mic configurations

File: clients/cpp-console/docs/RaspberryPi.md (hunk 5 above)
Comment (chschrae) [reply clients/cpp-console/docs/RaspberryPi.md]: Spelled it out. Is is Pi though not Pie. Pi the number not the dessert :)

File: clients/cpp-console/docs/RaspberryPi.md (hunk 4 above)
Comment (chschrae) [reply clients/cpp-console/docs/RaspberryPi.md]: fixed

**Pull request merged**